d2ir/compile.go

```diff
@@ -459,6 +461,12 @@ func (c *compiler) compileMap(dst *Map, ast, scopeAST *d2ast.Map) {
 						Value: []d2ast.InterpolationBox{{Substitution: n.Substitution}},
 					},
 				},
+				References: []*FieldReference{{
```

Comment:
out of curiosity why is this needed?

Reply:
It was just inconsistent that a Field existed without a ref. It was like an object without a pointer, so other parts of the code that looked for the pointers failed.